tests/scripts/thread-cert/Cert_6_1_02_REEDAttach_ED.py

```diff
@@ -77,10 +83,41 @@ def test(self):
         self.simulator.go(5)
         self.assertEqual(self.nodes[REED].get_state(), 'child')
 
-        self.nodes[ED].start()
-        self.simulator.go(5)
-        self.assertEqual(self.nodes[ED].get_state(), 'child')
+        self.nodes[MED].start()
+        self.simulator.go(config.DEFAULT_CHILD_TIMEOUT + 5)
+        self.assertEqual(self.nodes[MED].get_state(), 'child')
         self.assertEqual(self.nodes[REED].get_state(), 'router')
 
+        med_messages = self.simulator.get_messages_sent_by(MED)
+
+        # Step 2 - DUT sends MLE Parent Request
+        msg = med_messages.next_mle_message(mle.CommandType.PARENT_REQUEST)
+        self.assertEqual(0x02, msg.mle.aux_sec_hdr.key_id_mode)
+        check_parent_request(msg, is_first_request=True)
+
+        # Step 4 - DUT sends MLE Parent Request again
+        msg = med_messages.next_mle_message(mle.CommandType.PARENT_REQUEST)
+        self.assertEqual(0x02, msg.mle.aux_sec_hdr.key_id_mode)
+        check_parent_request(msg, is_first_request=False)
+
+        # Step 6 - DUT sends Child ID Request
+        msg = med_messages.next_mle_message(mle.CommandType.CHILD_ID_REQUEST)
+        check_child_id_request(msg, address_registration=CheckType.CONTAIN,
+            tlv_request=CheckType.CONTAIN, mle_frame_counter=CheckType.OPTIONAL,
+            route64=CheckType.OPTIONAL)
+        msg.assertSentToNode(self.nodes[REED])
+        check_tlv_request_tlv(msg, CheckType.CONTAIN, mle.TlvType.ADDRESS16)
+        check_tlv_request_tlv(msg, CheckType.CONTAIN, mle.TlvType.NETWORK_DATA)
```

Comment:
could we move the dest check and `check_tlv_request_tlv()` into `check_child_id_request()`?